### PR TITLE
Fluent-theme: GroupedList fluent styles

### LINKED ISSUE #7198: GroupedList: Fluent theme

### PR DESCRIPTION
**Pull request by Vitalius1**
#### Pull request checklist

- [X] Addresses an existing issue: Fixes #7198 
- [X] Include a change request file using `$ npm run change`

#### Description of changes
Adds fluent styles for GroupedList and adjust some minor things in `GroupSpacer` to facilitate some of the changes.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/7897)

### CODE REVIEW

File: packages/fluent-theme/src/fluent/styles/DetailsList.styles.ts [hunk 1]
```diff
@@ -41,5 +47,41 @@ export const DetailsRowStyles = (props: IDetailsRowStyleProps): Partial<IDetails
...
+      // in the DetailsList which takes into account the 40px default Check width vs the 48px fluent variation.
+      // When the fluent styles are to become the default ones this will be fixed in the logic.
+      width: detailsRowCheckWidth
+    }
+  };
+};
+
+// Using here any because those specific interfaces are missing from the public API. Consider tightening them in a separate PR.
+// tslint:disable-next-line:no-any
+export const DetailsHeaderStyles = (props: any): any => {
```
Comment (Vitalius1) [inline packages/fluent-theme/src/fluent/styles/DetailsList.styles.ts]: @KevinTCoughlin found out that DetailsHeader styles and styleProps interfaces are not part of the public API and the `IDetailsHeaderProps` are exported not through `index.ts` but through a series of re-exports all the way from `DetailsList.types.ts`. DetailsList and GroupedList need an API cleanup ASAP 😞 . Will leave this for now as `any`

File: packages/fluent-theme/src/fluent/styles/DetailsList.styles.ts [hunk 2]
```diff
@@ -41,5 +47,41 @@ export const DetailsRowStyles = (props: IDetailsRowStyleProps): Partial<IDetails
...
+    //   // Temporary using a hard coded value until fluent becomes default and we can access the value from the default constant.
+    //   marginRight: -12
+    // }
+  };
+};
+
+export const DetailsRowCheckStyles = (props: IDetailsRowCheckStyleProps): Partial<IDetailsRowCheckStyles> => {
+  return {
+    check: {
+      // Currently this causes a vertical scroll due to column width calculation logic
```
Comment (Jahnp) [inline packages/fluent-theme/src/fluent/styles/DetailsList.styles.ts]: Thank you for the detailed comments--these are really helpful for when these styles do eventually become the default in v7.

Comment (KevinTCoughlin) [reply packages/fluent-theme/src/fluent/styles/DetailsList.styles.ts]: Would it make sense to capture these comments as a diff which we can apply at a later date i.e. `7.x` release? This would remove the need for commented-out code while maintaining a reference in Github and a patch we can apply to adhere to new standards.

If we remove the related code in an isolated commit we can easily apply the inverse to a `7.x` branch.

File: packages/fluent-theme/src/fluent/styles/DetailsList.styles.ts (hunk 1 above)
Comment (KevinTCoughlin) [reply packages/fluent-theme/src/fluent/styles/DetailsList.styles.ts]: I'm willing to export those types to improve the public API now that we have a reason to. Is there anywhere else you've come across that may benefit? 

I'll take a look in the AM and see if I can commit the changes to your branch. cc: @natalieethell

File: packages/fluent-theme/src/fluent/styles/DetailsList.styles.ts [hunk 3]
```diff
@@ -41,5 +47,41 @@ export const DetailsRowStyles = (props: IDetailsRowStyleProps): Partial<IDetails
...
+
+// Using here any because those specific interfaces are missing from the public API. Consider tightening them in a separate PR.
+// tslint:disable-next-line:no-any
+export const DetailsHeaderStyles = (props: any): any => {
+  return {
+    // TODO: check with designer if needed according to relines.
+    // cellIsCheck: {
+    //   // Move the next cell after check to the left to align to redlines.
+    //   // Temporary using a hard coded value until fluent becomes default and we can access the value from the default constant.
+    //   marginRight: -12
```
Comment (KevinTCoughlin) [inline packages/fluent-theme/src/fluent/styles/DetailsList.styles.ts]: I'd take the commit / diff approach here to keep the source files clean of commented out code.

File: packages/office-ui-fabric-react/src/components/GroupedList/examples/GroupedList.Basic.Example.tsx [hunk 4]
```diff
@@ -86,6 +86,7 @@ export class GroupedListBasicExample extends React.Component<{}, IGroupedListExa
         selection={this._selection}
         selectionMode={SelectionMode.multiple}
         compact={this.state.isCompactMode}
+        styles={{ root: { width: '100%' } }} // expand the row all the way so it matches to the GroupHeader
```
Comment (KevinTCoughlin) [inline packages/office-ui-fabric-react/src/components/GroupedList/examples/GroupedList.Basic.Example.tsx]: I've actually tried to tackle this in the past re: the default case. You can see this problem on SharePoint document libraries... #6705

Comment (Vitalius1) [reply packages/office-ui-fabric-react/src/components/GroupedList/examples/GroupedList.Basic.Example.tsx]: So what was the reason the PR was not merged? I read all the comments but did not get why this was not a solution. Also, my opinion is that we should not do any changes in the source default styles of the `DetailsRow` to accommodate its use in the `GroupedList` example. When we use `GroupedList` outside the context of `DetailsList` and we just happened to use a `DetailsRow` as a renderer of the data cells we should just customize the styles of it to properly align with its headers. `DetailsRow` is a subcomponent of `DetailsList` and it's styled for its use within it. If you see this [example](https://developer.microsoft.com/en-us/fabric#/components/detailslist/grouped) you can see that the issue is not present as it has the necessary `viewport` value passed from its parent `DetailsList`.

So in short, if we decided to use `DetailsRow` outside of its initial intend we should provide custom styles so it can play well with the new parent. `DetailsRow` should not care or know that it's used within `GroupedList`. Let me know if my rationale is missing something and I am wrong on this one.

File: packages/fluent-theme/src/fluent/styles/DetailsList.styles.ts (hunk 3 above)
Comment (Vitalius1) [reply packages/fluent-theme/src/fluent/styles/DetailsList.styles.ts]: Sorry... I am not understanding what you mean when you say 

> capture these comments as a diff

I actually talked to Ben yesterday of this new alignment they designed for fluent styles and we came to the conclusion it needs another pass through to make sure we are not breaking anything else. Specifically this part in the toolkit where they don't have a 12px padding between the `Check` and the following cell in the attempt to align with the `GroupHeader`. But this requires changes in the DetailsList columns headers to align everything which I am hesitant to do as it might break some styling in OneDrive. The main issue is that the fluent styles provided by designers fro `DetailsList` and `GroupedList` are targeting mostly their use in OneDrive and not taken into consideration that these are Fabric components that can be used anywhere else. So I will get back to this comments as soon as we have a better approach in styling these parts.

**Comment by Vitalius1** [comment]
Labeling `Do not Merge` as there are some concerns that need to be discussed with designers before proceeding with it.

File: packages/fluent-theme/src/fluent/styles/DetailsList.styles.ts (hunk 1 above)
Comment (Vitalius1) [reply packages/fluent-theme/src/fluent/styles/DetailsList.styles.ts]: Thanks for the offer. Just let me know and I can push a change for that myself 👍 
Another place that would benefit is when using `onRenderGroupHeader` someone would like to provide some custom styles through the default renderer passed in the callback.

File: packages/fluent-theme/src/fluent/styles/DetailsList.styles.ts (hunk 3 above)
Comment (KevinTCoughlin) [reply packages/fluent-theme/src/fluent/styles/DetailsList.styles.ts]: My only suggestion was to stash these changes using git, perhaps even by pushing up to a remote branch and linking back in PR or code comment. This would remove the need for or at least the amount of in-line comments (if you want to refer to the commit itself as the comment you can). Then anyone can come thru at a later date, apply the commit, and raise a pull-request as a unit of work.

This is a cleaner approach as the commit will encapsulate the work for a point-in-time. With the comment approach, someone has to figure out what all needs to be changed which a commit does for you.

File: packages/office-ui-fabric-react/src/components/GroupedList/examples/GroupedList.Basic.Example.tsx (hunk 4 above)
Comment (KevinTCoughlin) [reply packages/office-ui-fabric-react/src/components/GroupedList/examples/GroupedList.Basic.Example.tsx]: Yea, it has to do with how the viewport is used which _does_ impact DetailsList in some cases. We can fix the example for the time-being, but I think this should be a default of the control. However, as I found out with the above PR there are a few edge cases to consider.

File: packages/fluent-theme/src/fluent/styles/DetailsList.styles.ts (hunk 1 above)
Comment (KevinTCoughlin) [reply packages/fluent-theme/src/fluent/styles/DetailsList.styles.ts]: Go ahead and push those changes if you have them ready. I would be happy to review them as part of this PR.

**Comment by Vitalius1** [comment]
Closing for now and reopen when having a clear path from designers on how to proceed with some of the concerns found.
